src/components/TodoModal/TodoModal.tsx

```
@@ -1,42 +1,72 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { useAppDispatch } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const dispatch = useAppDispatch();
+
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId)
```

Add error handling to the user fetch. Right now `getUser(todo.userId).then(setUser).finally(...)` swallows failures. Consider adding `.catch` to set an error state or clear the user (and optionally show a message in the modal). Example: `getUser(todo.userId).then(setUser).catch(() => setUser(null)).finally(() => setIsLoading(false))`. Also consider guarding against setState on unmounted component if needed.